src/domain/wallets/index.types.d.ts

```diff
@@ -12,11 +10,47 @@ type Deprecated = {
   readonly feeUsd: number
 }
 
+type InitiationViaIntraledger = {
+  readonly type: PaymentInitiationMethod["IntraLedger"]
+  readonly walletId: WalletId
+  readonly counterPartyUsername: Username
+}
+
+type InitiationViaLn = {
+  readonly type: PaymentInitiationMethod["Lightning"]
+  readonly paymentHash: PaymentHash
+  readonly pubkey: Pubkey
+}
+
+type InitiationViaOnChain = {
+  readonly type: PaymentInitiationMethod["OnChain"]
+  readonly address: OnChainAddress
+}
+
```

Comment:
```suggestion
FIXME: add OnChainAddress associated with old transaction to remove this legacy type
```

Reply:
I added this commit, but thinking about it now I'm not sure if we'd have this information anywhere else to backfill for transactions that were initiated onchain but settled intraledger 🤔